Help Center: Paginated search results

escapemanuele opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to p2rHU0-2iW-p2#comment-4238

## Proposed Changes

![image](https://github.com/user-attachments/assets/5c3c1e33-56b4-44cb-9ab3-77e9d79ed74c)

- We have now a bigger number of results coming from the search API, we are showing them all, with a "Search More"
- We hide "Recent Conversation" when searching.
- Changed "Recommended Resources" to "Search Results" when searching.
- "Show More" button is resetted when going through searches or removing the search query.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open the Help Center in various pages.
* Try searching for things like "DNS" or "Menu"

escapemanuele committed:
Paginated search results

fditrapani commented:
Nice work! this is looking good. We should update the spacing so the button is closer associated with the results and use the secondary button so that it's easier to notice:

<img width="1840" alt="image" src="https://github.com/user-attachments/assets/66fffdb5-6eee-4ec9-ab12-bd6d7029eb4d" />

escapemanuele committed:
Translation and variant

escapemanuele committed:
Different margin

escapemanuele commented:
Thanks @fditrapani, can you give it another try?

fditrapani commented:
Thanks for the update! A few things:

The show more button is showing for the recommended resources when I help the center and haven't done a search:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/1be81c72-55e4-4a4a-8d12-48e873009bc0" />

We can use some spacing after the show more button to better differentiate the results from the actions. I added 32px of margin to the top of the title:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/5a9559bc-8467-48f7-9425-82827d42b68f" />

If I do another search, it shows the expanded list vs resetting and going back to the initial 5:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/72adb117-1e51-4ce3-98f7-b7fdbce181c8" />

Also, somewhat related, we should hide the Recent conversation section when you type a search since you're focused on the search action. It can come back if you clear the search. It could also help to update the title from recommended resources to "Search results". What do you think?

escapemanuele committed:
Add margin on top

escapemanuele committed:
Reset visible result on query change

escapemanuele committed:
'Search Results' if searching

escapemanuele committed:
Show convos only if not searching

escapemanuele committed:
Fix number of visible results

escapemanuele commented:
> The show more button is showing for the recommended resources when I help the center and haven't done a search:

Well, yes, in that case you are in `/sites`, I see that we have 18 recommended resources in that case 😄 

> We can use some spacing after the show more button to better differentiate the results from the actions

Added 👍 

> If I do another search, it shows the expanded list vs resetting and going back to the initial 5:

Fixed!

> we should hide the Recent conversation section when you type a search ..........  update the title from recommended resources to "Search results"

Done!

fditrapani commented:
Thanks for the updates! this is working well now. A few minor things:

Now that we added the space between the resources, I feel like we need the same with the recent chats:
<img width="1138" alt="image" src="https://github.com/user-attachments/assets/a2cbd2c4-ca34-424f-b9f8-5893182b8871" />

Here is how that will look with the 32px margin added to the bottom of the recent chat:
<img width="1051" alt="image" src="https://github.com/user-attachments/assets/7a94c85c-fe72-4b06-a91d-7734de87b243" />

When I do a search, I still see the recent convos:
<img width="1332" alt="image" src="https://github.com/user-attachments/assets/2eea73be-5576-4fa7-b6ba-64f3651f5f52" />

We could also use the space above the show me where to section. Here's how that would look:
<img width="868" alt="image" src="https://github.com/user-attachments/assets/dd9ff09a-db65-4a41-9b86-2c4b1cf7e330" />

rcrdortiz commented on packages/help-center/src/components/help-center-search-results.tsx:
```diff
@@ -224,6 +225,12 @@ function HelpSearchResults( {
 	const searchResults = searchData ?? [];
 	const hasAPIResults = searchResults.length > 0;
 
+	const [ visibleResults, setVisibleResults ] = useState( 5 );
```
Maybe we could replace the magic number 5 with a constant, like this:
````
const DEFAULT_VISIBLE_RESULTS = 5;
````
This would make the code more readable and easier to maintain, especially if we decide to change the default number of visible results across the component in the future. What do you think?

If we move forward with this, we can update the rest of the code to reference the constant instead of using the number `5` directly.

escapemanuele replied:
That's a good idea 💯

escapemanuele committed:
Use constant

rcrdortiz commented:
Should we also include a `Show less` button so that the user can collapse to the previous state? Right now if I decide to expand the results I'm stuck at seeing all the results.

escapemanuele commented:
> Should we also include a Show less button so that the user can collapse to the previous state? Right now if I decide to expand the results I'm stuck at seeing all the results.

cc @fditrapani

escapemanuele commented:
Noticed that, I'm working on that in another PR 👍  @rcrdortiz https://linear.app/a8c/issue/DOTCOM-6426/help-center-0-results-search-does-not-provide-additional-help

fditrapani commented:
> Should we also include a Show less button so that the user can collapse to the previous state? Right now if I decide to expand the results I'm stuck at seeing all the results.

I personally don't think that's necessary but can be convinced if you all feel strongly about it :) 

> On searches with no results I'm getting a fully blank screen:


I noticed that the other day too and thought we could do this instead: 
![image](https://github.com/user-attachments/assets/283edd25-e069-4814-a317-4344e4440db2)

Clicking on ask AI could pass the search query over to Odie.

rcrdortiz commented:
This might just be me, having a button seems a bit odd. Do you think a link looks better than a button? Also, do you think that `Show All Results` is more accurate than `Show More`?
|Button|Link|
|-----|-----|
| ![Screenshot 2025-04-14 at 16 35 32](https://github.com/user-attachments/assets/79d53796-e6b5-4f1f-b856-f60110ae41a2)|![Screenshot 2025-04-14 at 16 37 14](https://github.com/user-attachments/assets/75f24f02-22b0-48e7-8cd5-4e52b8d8eedb)|

escapemanuele committed:
Add margin top to all h3

rcrdortiz reviewed (approved):
Besides the comments I've left, this PR works as expected and the code looks good to me.

escapemanuele commented:
> I noticed that the other day too and thought we could do this instead:

Updated the linear [issue](https://linear.app/a8c/issue/DOTCOM-6426/help-center-0-results-search-does-not-provide-additional-help), but at the moment we cannot pass queries to Odie @fditrapani 

> When I do a search, I still see the recent convos:

I'm not able to see it anymore, maybe it was an old live link? Can you try again?

fditrapani reviewed (approved):
Thanks for the updates! Looks great now. 

> This might just be me, having a button seems a bit odd. Do you think a link looks better than a button? 

The button offers a little more clarity and is easier not to miss. 

> Also, do you think that Show All Results is more accurate than Show More?

We're retrieving 5 results at a time so `Show more` is more accurate. If we showed everything, `Show all` would work. 

Side note, I just noticed the casing. We're moving to sentence case for everything. We'll need to update "Show More" > "Show more".

> Updated the linear [issue](https://linear.app/a8c/issue/DOTCOM-6426/help-center-0-results-search-does-not-provide-additional-help), but at the moment we cannot pass queries to Odie

No problem! Just opening Odie would be a good start!

fditrapani committed:
Fix button case
